js/util/combineCriteria.ts

```diff
@@ -0,0 +1,31 @@
+import * as Collections from 'typescript-collections';
+import { PropExtractorCriteria, PropExtractorConfig } from './extractProps';
+
+export const getCombinedCriteria = (
+  componentName: string,
+  configs: PropExtractorConfig[]
+): PropExtractorCriteria => {
+  let remainingConfigs: PropExtractorConfig[] = [...configs];
+  let inclusionSet = new Collections.Set<string>();
+
+  while (remainingConfigs.length > 0) {
+    let config = remainingConfigs.shift();
+    let criteria = config[componentName];
+
+    if (!criteria) {
+      continue;
+    }
+
+    let configInclusionSet = new Collections.Set<string>();
+    let configExclusionSet = new Collections.Set<string>();
+    criteria.include.map(s => configInclusionSet.add(s));
+    if (criteria.exclude) {
+      criteria.exclude.map(s => configExclusionSet.add(s));
+    }
+
+    inclusionSet.difference(configExclusionSet);
+    inclusionSet.union(configInclusionSet);
+  }
+
+  return { include: inclusionSet.toArray().sort() };
```

Comment:
Why does this need to be sorted?

Reply:
Since we're dealing with sets above, we don't get a consistent ordering.  This isn't strictly necessary but will make working with these things a little nicer in spots (testing, debug output, etc.) I could be convinced otherwise if you have a good reason.

Reply:
This just seems like YAGNI to me.  If we want ordering for testing/debug output/etc., the sorting should happen there, not here.

Reply:
Added a comment.